dev/com.ibm.ws.kernel.boot_fat/fat/src/com/ibm/ws/kernel/service/ServerEndpointControlMBeanTest.java

```diff
@@ -0,0 +1,280 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.kernel.service;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Set;
+
+import javax.management.JMX;
+import javax.management.MBeanException;
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectInstance;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.kernel.server.ServerEndpointControlMBean;
+//import com.ibm.websphere.kernel.server.ServerEndpointControlMBean;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import junit.framework.Assert;
+
+/**
+ * This test exercises the basic functionality of the ServerEndpointControlMBean by
+ * performing tests similar to those of the PauseResumeCommandTest except through the
+ * mbean interface instead of the server command
+ *
+ */
+@RunWith(FATRunner.class)
+public class ServerEndpointControlMBeanTest {
+
+    private static final String CONNECTOR_ADDRESS_FILE_NAME = "com.ibm.ws.jmx.local.address";
+    private static final String PAUSEABLE_EXCEPTION_CLASS = "com.ibm.ws.kernel.launch.service.PauseableComponentControllerRequestFailedException";
```

Comment:
Should PAUSEABLE_EXCEPTION_CLASS be changed to ...MBeanExcpetion?

Reply:
No, the server log will contain PauseableComponentControllerRequestFailedException from the PCC service component.  I don't bother to log the MBeanException as it contains the same info (it's basically a rethrow, but with using classes that aren't available on client side.